apps/backend/src/modules/auth/libs/types/auth-handler-options.type.ts

```diff
@@ -0,0 +1,8 @@
+import { type APIHandlerOptions } from "~/libs/modules/controller/controller.js";
+import { type UserResponseDto } from "~/modules/users/libs/types/types.js";
+
+type AuthHandlerOptions = APIHandlerOptions & {
+	user: UserResponseDto;
+};
```

Comment:
This type will be used only in one, I think we don't need to create a separate type for it. You can directly use `as APIHandlerOptions<{ user: UserResponseDto; }>`